workitem_blackbox_test.go

```diff
@@ -130,3 +135,96 @@ func TestListByFields(t *testing.T) {
 
 	test.DeleteWorkitemOK(t, nil, nil, controller, wi.ID)
 }
+
+func TestUnauthorizeCreateWI(t *testing.T) {
+	resource.Require(t, resource.Database)
+	// Build A token
+	token := jwt.New(jwt.SigningMethodRS256)
+	privatekey, err := jwt.ParseRSAPrivateKeyFromPEM(([]byte(RSAPrivateKeyTest)))
+	if err != nil {
+		t.Fatal("Could not parse Key ", err)
+	}
+	publickey, err := jwt.ParseRSAPublicKeyFromPEM(([]byte(RSAPublicKeyTest)))
+	if err != nil {
+		t.Fatal("Could not parse Key ", err)
+	}
+	tokenStr, err := token.SignedString(privatekey)
+	if err != nil {
+		t.Fatal("Could not sign the token ", err)
+	}
+
+	// Build a request
+	req, err := http.NewRequest("POST", "/api/workitems", nil)
+	if err != nil {
+		t.Fatal("could not create a HTTP request")
+	}
+
+	// Add Authorization Header in wrong format
+	authToken := fmt.Sprintf("WRONG TOKEN Format Bearer %s", tokenStr)
+	req.Header.Add("Authorization", authToken)
+
+	rr := httptest.NewRecorder()
+	ts := models.NewGormTransactionSupport(DB)
+	wir := models.NewWorkItemTypeRepository(ts)
+	repo := models.NewWorkItemRepository(ts, wir)
+
+	// temperory service for testing the middleware
+	service := goa.New("TestUnauthorizedCreateWI-Service")
+	assert.NotNil(t, service)
+
+	// append a middleware to service. Use appropriate RSA keys
+	jwtMiddleware := goajwt.New(publickey, nil, app.NewJWTSecurity())
+	app.UseJWTMiddleware(service, jwtMiddleware)
+
+	controller := NewWorkitemController(service, repo, ts)
+	app.MountWorkitemController(service, controller)
+
+	// Hit the service with own request
+	service.Mux.ServeHTTP(rr, req)
+
+	// the problem I am facing right now is
+	// When no token is set in Authorization header, it throws -- uncaught error err=[NNCptvpI] 401 jwt_security_error: missing header "Authorization"
+	// When wrong formatted token is set in header, it throws -- 401 jwt_security_error: invalid or malformed
+	// This is as expected till now,
+	// But problem = it `throws error` and prints out as `uncaught_error`
+	// Hence I can not capture response status code. any suggestions ?
+	fmt.Println(rr.Code) // always prints 500 because of uncaught_error
```

Comment:
Yes @sbose78 , do you mean test should do `assert.NotEqual(rr.Code, 201)`
201 because it is Create call, it will change as per intended API under test.